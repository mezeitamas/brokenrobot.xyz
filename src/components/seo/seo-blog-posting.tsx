import React from 'react';
import type { FunctionComponent, PropsWithChildren } from 'react';

import { useSiteMetadata } from '../../hooks/use-site-metadata';

import { MetaOgArticle } from './meta-og-article/meta-og-article';
import { MetaTwitter } from './meta-twitter/meta-twitter';
import { RichResultsBlogPosting } from './rich-results-blog-posting/rich-results-blog-posting';

type SeoBlogPostingProps = {
    title: string;
    description: string;
    pathname: string;
    published: string;
    tags: string[];
};

const SeoBlogPosting: FunctionComponent<PropsWithChildren<SeoBlogPostingProps>> = ({
    title,
    description,
    pathname,
    published,
    tags,
    children
}) => {
    const { title: siteName, siteUrl, author } = useSiteMetadata();
    const url = `${siteUrl}${pathname}`;

    return (
        <>
            <html lang="en" />

            <title>{title}</title>

            <meta
                httpEquiv="cache-control"
                content="public, max-age=0, must-revalidate"
            />

            <meta
                name="description"
                content={description}
            />

            <meta
                name="keywords"
                content={tags.join(', ')}
            />

            <MetaOgArticle
                siteName={siteName}
                title={title}
                description={description}
                url={url}
                published={published}
                tags={tags}
                author={author.name}
            />

            <RichResultsBlogPosting
                siteName={siteName}
                siteUrl={siteUrl}
                title={title}
                description={description}
                url={url}
                published={published}
                author={author}
            />

            <MetaTwitter
                title={title}
                description={description}
                url={url}
            />

            {children}
        </>
    );
};

export { SeoBlogPosting };

export type { SeoBlogPostingProps };
