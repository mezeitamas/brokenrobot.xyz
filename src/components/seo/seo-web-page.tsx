import React from 'react';
import type { FunctionComponent, PropsWithChildren } from 'react';

import { useSiteMetadata } from '../site-metadata/use-site-metadata';

import { MetaOgWebsite } from './meta-og-website/meta-og-website';
import { MetaTwitter } from './meta-twitter/meta-twitter';
import { RichResultsWebPage } from './rich-results-web-page/rich-results-web-page';

type SeoWebPageProps = {
    title: string;
    description: string;
    pathname: string;
};

const SeoWebPage: FunctionComponent<PropsWithChildren<SeoWebPageProps>> = ({
    title,
    description,
    pathname,
    children
}) => {
    const { title: siteName, siteUrl, author } = useSiteMetadata();
    const url = `${siteUrl}${pathname}`;

    return (
        <>
            <html lang="en" />

            <title>{title}</title>

            <meta
                httpEquiv="cache-control"
                content="public, max-age=0, must-revalidate"
            />

            <meta
                name="description"
                content={description}
            />

            <MetaOgWebsite
                siteName={siteName}
                title={title}
                description={description}
                url={url}
            />

            <RichResultsWebPage
                siteName={siteName}
                siteUrl={siteUrl}
                author={author}
            />

            <MetaTwitter
                title={title}
                description={description}
                url={url}
                author={author}
            />

            {children}
        </>
    );
};

export { SeoWebPage };

export type { SeoWebPageProps };
