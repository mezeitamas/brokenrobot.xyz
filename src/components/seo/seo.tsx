import React from 'react';
import type { FunctionComponent, PropsWithChildren } from 'react';

import { useSiteMetadata } from '../../hooks/use-site-metadata';

import { MetaOgArticle } from './meta-og-article/meta-og-article';
import { MetaOgWebsite } from './meta-og-website/meta-og-website';
import { MetaTwitter } from './meta-twitter/meta-twitter';
import { RichResultsArticle } from './rich-results-article/rich-results-article';

type SeoProps = {
    title: string;
    description: string;
    pathname: string;
    isArticle: boolean;
    published?: string;
};

const Seo: FunctionComponent<PropsWithChildren<SeoProps>> = ({
    title,
    description,
    pathname,
    isArticle,
    published,
    children
}) => {
    const { title: siteName, siteUrl, author } = useSiteMetadata();
    const url = `${siteUrl}${pathname}`;

    return (
        <>
            <html lang="en" />

            <title>{title}</title>

            <meta
                httpEquiv="cache-control"
                content="public, max-age=0, must-revalidate"
            />

            <meta
                name="description"
                content={description}
            />

            {isArticle === true ? (
                <>
                    <MetaOgArticle
                        siteName={siteName}
                        title={title}
                        description={description}
                        url={url}
                        published={published === undefined ? '' : published}
                        author={author.name}
                    />

                    <RichResultsArticle
                        siteName={siteName}
                        siteUrl={siteUrl}
                        title={title}
                        url={url}
                        published={published === undefined ? '' : published}
                        author={author.name}
                    />
                </>
            ) : null}

            {isArticle !== true ? (
                <MetaOgWebsite
                    siteName={siteName}
                    title={title}
                    description={description}
                    url={url}
                />
            ) : null}

            <MetaTwitter
                title={title}
                description={description}
                url={url}
            />

            {children}
        </>
    );
};

export { Seo };

export type { SeoProps };
