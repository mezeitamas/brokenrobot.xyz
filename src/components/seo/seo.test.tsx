import React from 'react';

import { create } from 'react-test-renderer';

import { Seo } from './seo';
import type { SeoProps } from './seo';

jest.mock('../../hooks/use-site-metadata', () => {
    return {
        useSiteMetadata: jest.fn().mockImplementation(() => {
            return {
                title: 'Broken Robot',
                description:
                    'Personal website and blog of Tamas Mezei. Welcome to my little corner of the web, where I share my professional experiences, thoughts, adventures, and projects with the world.',
                siteUrl: 'https://www.brokenrobot.xyz',
                author: {
                    name: 'Tamas Mezei',
                    social: {
                        github: 'https://github.com/mezeitamas',
                        linkedin: 'https://www.linkedin.com/in/mezeitamas/'
                    }
                }
            };
        })
    };
});

describe('<Seo />', () => {
    test('should render website', () => {
        // Given
        const props: SeoProps = {
            title: 'Blog',
            description: 'Blog posts',
            pathname: '/blog/',
            isArticle: false
        };

        // When
        const component = create(<Seo {...props} />).toJSON();

        // Then
        expect(component).toMatchSnapshot();
    });

    test('should render article', () => {
        // Given
        const props: SeoProps = {
            title: 'Hosting a static website on Amazon S3',
            description:
                'Hosting a simple static website on Amazon Web Services (AWS) can be daunting at first, but luckily itâ€™s quite straightforward. One option...',
            pathname: '/blog/hosting-a-static-website-on-amazon-s3/',
            isArticle: true,
            published: '2023-05-04'
        };

        // When
        const component = create(<Seo {...props} />).toJSON();

        // Then
        expect(component).toMatchSnapshot();
    });
});
