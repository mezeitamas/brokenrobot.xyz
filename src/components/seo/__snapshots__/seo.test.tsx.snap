// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Seo /> should render with props 1`] = `
[
  <html
    lang="en"
  />,
  <title>
    Hosting a static website on Amazon S3
  </title>,
  <meta
    content="public, max-age=0, must-revalidate"
    httpEquiv="cache-control"
  />,
  <meta
    content="Hosting a simple static website on Amazon Web Services (AWS) can be daunting at first, but luckily it’s quite straightforward. One option..."
    name="description"
  />,
  <meta
    content="Hosting a static website on Amazon S3"
    property="og:title"
  />,
  <meta
    content="Hosting a simple static website on Amazon Web Services (AWS) can be daunting at first, but luckily it’s quite straightforward. One option..."
    property="og:description"
  />,
  <meta
    content="website"
    property="og:type"
  />,
  <meta
    content="Hosting a static website on Amazon S3"
    name="twitter:title"
  />,
  <meta
    content="Hosting a simple static website on Amazon Web Services (AWS) can be daunting at first, but luckily it’s quite straightforward. One option..."
    name="twitter:description"
  />,
  <meta
    content="summary"
    name="twitter:card"
  />,
  <meta
    content="https://www.brokenrobot.xyzhosting-a-static-website-on-amazon-s3"
    name="twitter:url"
  />,
]
`;

exports[`<Seo /> should render without props 1`] = `
[
  <html
    lang="en"
  />,
  <title>
    Broken Robot
  </title>,
  <meta
    content="public, max-age=0, must-revalidate"
    httpEquiv="cache-control"
  />,
  <meta
    content="Personal website and blog of Tamas Mezei. Welcome to my little corner of the web, where I share my professional experiences, thoughts, adventures, and projects with the world."
    name="description"
  />,
  <meta
    content="Broken Robot"
    property="og:title"
  />,
  <meta
    content="Personal website and blog of Tamas Mezei. Welcome to my little corner of the web, where I share my professional experiences, thoughts, adventures, and projects with the world."
    property="og:description"
  />,
  <meta
    content="website"
    property="og:type"
  />,
  <meta
    content="Broken Robot"
    name="twitter:title"
  />,
  <meta
    content="Personal website and blog of Tamas Mezei. Welcome to my little corner of the web, where I share my professional experiences, thoughts, adventures, and projects with the world."
    name="twitter:description"
  />,
  <meta
    content="summary"
    name="twitter:card"
  />,
  <meta
    content="https://www.brokenrobot.xyz"
    name="twitter:url"
  />,
]
`;
